lambda/enable_guardduty.py

```diff
@@ -0,0 +1,330 @@
+    response = gd_client.list_members(DetectorId=detector_id, MaxResults=50)
+    while 'NextToken' in response:
+        for a in response['Members']:
+            # Convert to a lookup table
+            output[a['AccountId']] = a
+        response = gd_client.list_members(
+            DetectorId=detector_id,
+            MaxResults=50,
+            NextToken=response['NextToken'],
+        )
+    for a in response['Members']:
+        # Convert to a lookup table
+        output[a['AccountId']] = a
+
+    return(output)
+
+
+def invite_account(account, detector_id, gd_client, event, region):
+    if event["dry_run"]:
+        logger.info(f"Need to Invite {account['Name']}({account['Id']}) to this GuardDuty Master")
+        return(None)
+
+    logger.info(f"Inviting {account['Name']}({account['Id']}) to this GuardDuty Master")
+    gd_client.create_members(
+        AccountDetails=[
+            {
+                'AccountId': account['Id'],
+                'Email': account['Email'],
+            },
+        ],
+        DetectorId=detector_id,
+    )
+    gd_client.invite_members(
+        AccountIds=[account['Id']],
+        DetectorId=detector_id,
+        Message=event["message"],
+    )
+
+
+def accept_invite(account, role_name, event, region):
+    if event["dry_run"]:
+        logger.info(f"Need to accept invite in {account['Name']}({account['Id']})")
+        return(None)
+
+    logger.info(f"Accepting invite in {account['Name']}({account['Id']})")
+
+    role_arn = create_role_arn(account['Id'], role_name)
+    creds = get_creds(role_arn)
+    if creds is False:
+        logger.error(
+            f"Unable to assume role into {account['Name']}({account['Id']}) to accept invite")
+        return(False)
+
+    child_client = boto3.client(
+        'guardduty',
+        region_name=region,
+        aws_access_key_id=creds['AccessKeyId'],
+        aws_secret_access_key=creds['SecretAccessKey'],
+        aws_session_token=creds['SessionToken'],
+    )
+
+    response = child_client.list_detectors()
+    if len(response['DetectorIds']) == 0:
+        response = child_client.create_detector(Enable=True)
+        detector_id = response['DetectorId']
+    else:
+        detector_id = response['DetectorIds'][0]
+
+    response = child_client.list_invitations()
+    for i in response['Invitations']:
+        child_client.accept_invitation(
+            DetectorId=detector_id,
+            InvitationId=i['InvitationId'],
+            MasterId=i['AccountId'],
+        )
+
+
+def create_role_arn(account_id, role_name):
+    return f"arn:aws:iam::{account_id}:role/{role_name}"
+
+
+def get_creds(role_arn):
+    client = boto3.client('sts')
+    try:
+        session = client.assume_role(
+            RoleArn=role_arn, RoleSessionName="EnableGuardDuty",
+        )
+        return(session['Credentials'])
+    except Exception as e:
+        logger.error(f"Failed to assume role {role_arn}: {e}")
+        return(False)
+
+
+def describe_account(event):
```

Comment:
If we called this from the handler vs process_region() the lambda might run marginally faster